packages/client/wallets/smart-wallet/src/blockchain/wallets/SendTransactionService.ts

```diff
@@ -0,0 +1,195 @@
+import { SmartAccountClient } from "permissionless";
+import { EntryPoint } from "permissionless/_types/types";
+import { SmartAccount } from "permissionless/accounts";
+import {
+    Abi,
+    Address,
+    BaseError,
+    Chain,
+    ContractFunctionRevertedError,
+    Hex,
+    PublicClient,
+    TransactionReceipt,
+    Transport,
+} from "viem";
+
+import { CrossmintSDKError, WalletErrorCode } from "@crossmint/client-sdk-base";
+
+export type TransactionServiceTransactionRequest = {
+    address: Address;
+    abi: Abi;
+    functionName: string;
+    args: any;
+    value?: bigint;
+};
+
+/**
+ * Error thrown when a transaction fails to send.
+ * @param viemError The error thrown by the viem client. See https://viem.sh/docs/glossary/errors.html
+ */
+export class EVMSendTransactionError extends CrossmintSDKError {
+    constructor(message: string, public readonly viemError: BaseError, code = WalletErrorCode.SEND_TRANSACTION_FAILED) {
+        super(message, code);
+    }
+}
```

Comment:
@bryan-eastwood3 what do you see as the main value prop for having these extra errors, compared to letting the viem ones bubble up?

Reply:
Is there any comparisons somewhere?

Reply:
Two main things:

- If we just bubble up the raw viem error, we can't add any functionality, e.g. if we wanted to always include a tenderly link in the error
- We generally have more ability to optimize the experience for long tail devs, e.g. by including the viem revert error as a field in our revert error a dev won't have to call `walk` and get it themselves (obviously very basic but we want to make this flow very frictionless). Another example would be us wrapping errors in different subclasses that indicate at which point the transaction failed (`EVMSendTransactionConfirmationError` for confirmation stage, `EVMSendTransactionSimulationError` for simulation)